samples/matter/bridge/src/bridge_manager.cpp

```diff
@@ -0,0 +1,191 @@
+	case BridgedDevice::DeviceType::HumiditySensor: {
+		LOG_INF("Adding HumiditySensor bridged device");
+		device = Platform::New<HumiditySensorDevice>(nodeLabel);
+		break;
+	}
+#endif /* CONFIG_BRIDGE_HUMIDITY_SENSOR_BRIDGED_DEVICE */
+	default:
+		return CHIP_ERROR_INVALID_ARGUMENT;
+	}
+
+	if (device == nullptr || device->mDataVersion == nullptr) {
+		return CHIP_ERROR_NO_MEMORY;
+	}
+
+	CHIP_ERROR err = AddDevice(device);
+
+	/* In case adding failed, free the allocated memory for the device instance. */
+	if (err != CHIP_NO_ERROR) {
+		Platform::Delete(device);
+	}
+
+	return err;
+}
+
+CHIP_ERROR BridgeManager::RemoveBridgedDevice(uint16_t endpoint)
+{
+	uint8_t index = 0;
+
+	while (index < kMaxBridgedDevices) {
+		if (nullptr != mBridgedDevices[index]) {
+			if (mBridgedDevices[index]->GetEndpointId() == endpoint) {
+				LOG_INF("Removed dynamic endpoint %d (index=%d)", endpoint, index);
+				/* Free dynamically allocated memory */
+				emberAfClearDynamicEndpoint(index);
+				Platform::Delete(mBridgedDevices[index]);
+				mBridgedDevices[index] = nullptr;
+				return CHIP_NO_ERROR;
+			}
+		}
+		index++;
+	}
+
+	return CHIP_ERROR_NOT_FOUND;
+}
+
+CHIP_ERROR BridgeManager::AddDevice(BridgedDevice *device)
+{
+	uint8_t index = 0;
+
+	while (index < kMaxBridgedDevices) {
+		/* Find the first empty index in the brdiged devices list */
+		if (nullptr == mBridgedDevices[index]) {
+			EmberAfStatus ret;
+			while (true) {
+				ret = emberAfSetDynamicEndpoint(
+					index, mCurrentDynamicEndpointId, device->mEp,
+					Span<DataVersion>(device->mDataVersion, device->mDataVersionSize),
+					Span<const EmberAfDeviceType>(device->mDeviceTypeList,
+								      device->mDeviceTypeListSize));
+
+				if (ret == EMBER_ZCL_STATUS_SUCCESS) {
+					LOG_INF("Added device to dynamic endpoint %d (index=%d)",
+						mCurrentDynamicEndpointId, index);
+					mBridgedDevices[index] = device;
+					device->Init(mCurrentDynamicEndpointId);
+					return CHIP_NO_ERROR;
+				} else if (ret != EMBER_ZCL_STATUS_DUPLICATE_EXISTS) {
+					LOG_ERR("Failed to add dynamic endpoint: Internal error!");
+					return CHIP_ERROR_INTERNAL;
+				}
+
+				/* Handle wrap condition */
+				if (++mCurrentDynamicEndpointId < mFirstDynamicEndpointId) {
+					mCurrentDynamicEndpointId = mFirstDynamicEndpointId;
+				}
+			}
+		}
+		index++;
+	}
+
+	LOG_ERR("Failed to add dynamic endpoint: No endpoints available!");
+
+	return CHIP_ERROR_NO_MEMORY;
+}
+
+CHIP_ERROR BridgeManager::HandleRead(uint16_t endpoint, ClusterId clusterId,
+				     const EmberAfAttributeMetadata *attributeMetadata, uint8_t *buffer,
+				     uint16_t maxReadLength)
+{
+	BridgedDevice *device = sBridgeManager.mBridgedDevices[endpoint];
+	if ((endpoint >= kMaxBridgedDevices) || (device == nullptr) || !buffer || !attributeMetadata) {
+		return CHIP_ERROR_INVALID_ARGUMENT;
+	}
+
+	return device->HandleRead(device, clusterId, attributeMetadata->attributeId, buffer, maxReadLength);
```

Comment:
You don't need to pass `device` as the first argument if it's already used as `this`.